materialize-elasticsearch/transactor.go

```diff
@@ -187,10 +199,25 @@ func (t *transactor) Store(it *m.StoreIterator) (m.StartCommitFunc, error) {
 		lastIndex = b.index
 
 		id := base64.RawStdEncoding.EncodeToString(it.PackedKey)
+
+		var routingVal []byte
+		if b.routingField != nil {
+			routing := it.Key[*b.routingField]
+			var err error
+			routingVal, err = json.Marshal(routing)
+			if err != nil {
+				return nil, fmt.Errorf("marshaling routing value %v: %w", routing, err)
+			}
+		}
+
 		if it.Delete && t.cfg.HardDelete {
 			// Ignore items which do not exist and are already deleted.
 			if it.Exists {
-				batch = append(batch, []byte(`{"delete":{"_id":"`+id+`"}}`)...)
+				batch = append(batch, []byte(`{"delete":{"_id":"`+id+`"`)...)
+				if routingVal != nil {
+					batch = append(batch, []byte(`,"routing":`+string(routingVal))...)
```

Comment:
nit: I'm pretty sure the compiler isn't smart enough to avoid the extra allocation / copies here, as compared to:

```
batch = append(append(batch, `,"routing":"`...), routingVal...)
```